src/character.cpp

```diff
@@ -12343,10 +12348,10 @@ stat_mod Character::get_pain_penalty() const
 
     // Prevent negative penalties, there is better ways to give bonuses for pain
     // Also not make character has 0 stats
-    ret.strength = std::max( ret.strength, 1 );
-    ret.dexterity = std::max( ret.dexterity, 1 );
-    ret.intelligence = std::max( ret.intelligence, 1 );
-    ret.perception = std::max( ret.perception, 1 );
+    ret.strength = std::clamp( ret.strength, 1, get_str() - 1 );
```

Comment:
@osuphobia this introduces an assertion failure when stats fall too low, in libstdc++ when assertions are turned on, because the `hi` value can be equal or less than `lo` as arguments to clamp, and it is undefined behavior (so glibcxx asserts).

```cpp
gdb$ bt
#0  0x00007f28862b01dc in ?? () from /usr/lib64/libc.so.6
#1  0x00007f28862614f2 in raise () from /usr/lib64/libc.so.6
#2  0x00007f288624a4ed in abort () from /usr/lib64/libc.so.6
#3  0x00007f28864d69f9 in std::__glibcxx_assert_fail (
    file=file@entry=0x5597dca4bcd0 "/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/stl_algo.h", line=line@entry=0xe55,
    function=function@entry=0x5597dc9f0380 "constexpr const _Tp& std::clamp(const _Tp&, const _Tp&, const _Tp&) [with _Tp = int]",
    condition=condition@entry=0x5597dc8f2ce2 "!(__hi < __lo)")
    at /var/tmp/portage/sys-devel/gcc-13.3.0/work/gcc-13.3.0/libstdc++-v3/src/c++11/assert_fail.cc:41
#4  0x00005597dce576cf in std::clamp<int> (__val=@0x7ffe7b6c8d10: 0x2, __lo=<optimized out>, __hi=<optimized out>)
    at /usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/stl_algo.h:3667
#5  std::clamp<int> (__hi=<optimized out>, __lo=<optimized out>, __val=<optimized out>)
    at /usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/stl_algo.h:3667
#6  Character::get_pain_penalty (this=this@entry=0x5597e2d850e0) at ../src/character.cpp:12351
#7  0x00005597dce97cfc in Character::recalc_speed_bonus (this=0x5597e2d850e0) at ../src/character.cpp:12224
#8  0x00005597dd136985 in Creature::process_turn (this=0x5597e2d850e0) at ../src/creature.cpp:368
#9  Creature::process_turn (this=this@entry=0x5597e2d850e0) at ../src/creature.cpp:355
#10 0x00005597dce9f1e6 in Character::process_turn (this=0x5597e2d850e0) at ../src/character.cpp:2330
#11 0x00005597dd298d32 in do_turn () at ../src/do_turn.cpp:675
#12 0x00005597dcabe7b5 in main (argc=<optimized out>, argv=<optimized out>) at ../src/main.cpp:873
```

Reply:
```suggestion
    ret.strength = get_str() > 2 ? std::clamp( ret.strength, 1, get_str() - 1 ) : std::clamp( ret.strength, 0, 1 );
```
So I should add a check like this to prevent the issue?